packages/apps/plugins/plugin-markdown/src/components/DocumentEditor.tsx

```diff
@@ -6,28 +6,73 @@ import { EditorView } from '@codemirror/view';
 import React, { useEffect, useMemo } from 'react';
 
 import { type DocumentType } from '@braneframe/types';
+import { useResolvePlugin, parseFileManagerPlugin, useIntentDispatcher } from '@dxos/app-framework';
 import { createDocAccessor, fullyQualifiedId, getSpace } from '@dxos/react-client/echo';
 import { useIdentity } from '@dxos/react-client/halo';
-import { createDataExtensions, listener, localStorageStateStoreAdapter, state } from '@dxos/react-ui-editor';
+import {
+  createDataExtensions,
+  listener,
+  localStorageStateStoreAdapter,
+  state,
+  type Extension,
+} from '@dxos/react-ui-editor';
 
-import EditorMain, { type EditorMainProps } from './EditorMain';
+import MarkdownEditor, { type MarkdownEditorProps } from './MarkdownEditor';
+import { getBaseExtensions } from '../extensions';
+import { type MarkdownPluginState, type MarkdownSettingsProps } from '../types';
 import { getFallbackName, setFallbackName } from '../util';
 
-type DocumentEditorProps = { document: DocumentType } & Omit<EditorMainProps, 'id'>;
+type DocumentEditorProps = {
+  document: DocumentType;
+  settings: MarkdownSettingsProps;
+} & Omit<MarkdownEditorProps, 'id' | 'inputMode' | 'toolbar' | 'extensions'> &
+  Pick<MarkdownPluginState, 'extensionProviders'>;
 
 /**
  * Editor for a `DocumentType`.
  */
 // TODO(wittjosiah): Reconcile with DocumentSection & DocumentCard.
```

Comment:
This can be removed now (or maybe moved over to the card component, which should be removed eventually.